File: frontend/lib/models/auth_response.dart
class AuthResponse {
  final int userId;
  final String email;
  final String displayName;

  AuthResponse({
    required this.userId,
    required this.email,
    required this.displayName,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      userId: json['userId'] as int,
      email: json['email'] as String,
      displayName: json['displayName'] as String,
    );
  }
}

File: backend/src/main/java/assistant_coaching/demo/dto/AuthResponse.java
package assistant_coaching.demo.dto;

public class AuthResponse {

    private Long userId;
    private String email;
    private String displayName;

    public AuthResponse() {
    }

    public AuthResponse(Long userId, String email, String displayName) {
        this.userId = userId;
        this.email = email;
        this.displayName = displayName;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }
}
